src/optimizer/simple_optimizer.cpp

```diff
@@ -1014,26 +1006,14 @@ SimpleOptimizer::CreateHackingNestedLoopJoinPlan(
   auto projection = CreateHackProjection();
   auto schema = CreateHackJoinSchema();
 
-  // LEFT.4 == RIGHT.1
-  expression::TupleValueExpression* left_table_attr_4 =
-      new expression::TupleValueExpression(type::Type::INTEGER, 0, 0);
-
-  expression::TupleValueExpression* right_table_attr_1 =
-      new expression::TupleValueExpression(type::Type::INTEGER, 1, 0);
-
-  std::unique_ptr<const expression::AbstractExpression> join_predicate(
-      new expression::ComparisonExpression(EXPRESSION_TYPE_COMPARE_EQUAL,
-                                           left_table_attr_4,
-                                           right_table_attr_1));
-
   std::vector<oid_t> join_column_ids_left = {0};   // I_I_ID in the result
   std::vector<oid_t> join_column_ids_right = {1};  // S_I_ID in the table
```

Comment:
So instead of storing the physical offset in the table of the right child operator, I think it's better to store the logical offset of the output schema here as well, just as with the left child operator. In that case the two variables are consistent, and I think it's better to unify the logic for both index NestedLoopJoin and seq NestedLoopJoin. Otherwise in the seqscan version of NestedLoopJoin, you have to know the schema of the base table of the right child operator to perform the join here correctly, which is not desirable.

Well if you store the logical offset, you can still get the physical offset in the index_scan_plan with IndexScanPlan::column_ids_. The only disadvantage here is that we have to select the join column of the right child to be one of its output schema columns, even though sometimes we don't need to output the join column. But I think returning one extra column here is acceptable (we have to scan it anyway).